go-controller/pkg/libovsdbops/router.go

```diff
@@ -313,6 +313,24 @@ func CreateOrUpdateLogicalRouterPolicyWithPredicateOps(nbClient libovsdbclient.C
 	return m.CreateOrUpdateOps(ops, opModels...)
 }
 
+func CreateOrUpdateLogicalRouterPoliciesOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation,
+	lrps ...*nbdb.LogicalRouterPolicy) ([]libovsdb.Operation, error) {
+	opModels := make([]operationModel, 0, len(lrps))
+	for i := range lrps {
+		lrp := lrps[i]
+		opModel := operationModel{
+			Model:          lrp,
+			OnModelUpdates: onModelUpdatesAllNonDefault(),
+			ErrNotFound:    false,
+			BulkOp:         false,
+		}
+		opModels = append(opModels, opModel)
+	}
+
+	m := newModelClient(nbClient)
+	return m.CreateOrUpdateOps(ops, opModels...)
+}
+
```

Comment:
I don't think I want to actually ever create LRP policies with this function, so what is the right way to make it Update-only (not sure if this is possible?)

Reply:
You just need to set `ErrNotFound` to `true`.

But a CreateOrUpdate would be more re-usable. No chance to pass the router these LRPs are associated with?

Reply:
I think I know the router, but I don't want this code to create lrp if it doesn't exist. Also if I set ErrNotFound to true I don't think we want to fail because of that error, so should I just if error is not found and ignore it?

Reply:
Yup.

Why the specific concern with LRPs? You are already using `CreateOrUpdateACLsOps` and `CreateOrUpdateQoSesOps` in a similar situation in the syncer.

Reply:
I guess because acls without a reference to the switch/port group will be deleted :) 
it's probably the same for qos and lrp

Reply:
This is a bit nuanced (why it would be better to have a router here)

`CreateOrUpdateQoSesOps` is being used to associate the same QoS to multiple switches so it makes sense for it to no being provided with a switch. This is quite a novelty.

`CreateOrUpdateACLsOps` is slightly different because they can either be associated to a switch or a PG, so the user always has the chance to do that in a later op.

`CreateOrUpdateACLs` is actually wrong, because that will never be able to create ACLs, only update them.

Although `CreateOrUpdateLogicalRouterPoliciesOps` could be used as `CreateOrUpdateQoSesOps`, that is to associate the same LRP to multiple routers, we are already assuming we don't do this because on the delete we delete from the router and we also delete the LRPs themselves, something it wouldn't work were they associated to multiple router. So that it why it makes sense to take care in CreateOrUpdateLogicalRouterPoliciesOps to associate them to a router.

Reply:
So if we follow this analogy, the initial implementation of `CreateOrUpdateLogicalRouterPoliciesOps` is correct?

Reply:
If you thin a Update specific method is better to work with, no problem with that either.